test/facility-guarantee/post-facility-guarantee.api-test.ts

```diff
@@ -0,0 +1,302 @@
+      guaranteeTypeCode,
+    },
+  ];
+
+  let api: Api;
+
+  beforeAll(async () => {
+    api = await Api.create();
+  });
+
+  afterAll(async () => {
+    await api.destroy();
+  });
+
+  afterEach(() => {
+    nock.abortPendingRequests();
+    nock.cleanAll();
+  });
+
+  const { idToken, givenAuthenticationWithTheIdpSucceeds } = withAcbsAuthenticationApiTests({
+    givenRequestWouldOtherwiseSucceed: () => givenRequestToCreateFacilityGuaranteeInAcbsSucceeds(),
+    makeRequest: () => api.post(createFacilityGuaranteeUrl, requestBodyToCreateFacilityGuarantee),
+    successStatusCode: 201,
+  });
+
+  withClientAuthenticationTests({
+    givenTheRequestWouldOtherwiseSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenRequestToCreateFacilityGuaranteeInAcbsSucceeds();
+    },
+    makeRequestWithoutAuth: (incorrectAuth?: IncorrectAuthArg) =>
+      api.postWithoutAuth(createFacilityGuaranteeUrl, requestBodyToCreateFacilityGuarantee, incorrectAuth?.headerName, incorrectAuth?.headerValue),
+  });
+
+  it('returns a 201 response with the facility guarantee location if it has been successfully created in ACBS', async () => {
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequest = givenRequestToCreateFacilityGuaranteeInAcbsSucceeds();
+
+    const { status, body } = await api.post(createFacilityGuaranteeUrl, requestBodyToCreateFacilityGuarantee);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      facilityIdentifier,
+    });
+    expect(acbsRequest.isDone()).toBe(true);
+  });
+
+  it('rounds the maximumLiability to 2dp', async () => {
+    const requestBodyWithMaximumLiabilityToRound = [{ ...requestBodyToCreateFacilityGuarantee[0], maximumLiability: 1.234 }];
+    const acbsRequestBodyWithRoundedMaximumLiability = {
+      ...acbsRequestBodyToCreateFacilityGuarantee,
+      GuaranteedLimit: 1.23,
+    };
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequestWithRoundedMaximumLiability = requestToCreateFacilityGuaranteeInAcbsWithBody(acbsRequestBodyWithRoundedMaximumLiability).reply(
+      201,
+      undefined,
+      {
+        location: `/Portfolio/${portfolioIdentifier}/Facility/${facilityIdentifier}/FacilityGuarantee?accountOwnerIdentifier=00000000&lenderTypeCode=${lenderTypeCode}&sectionIdentifier=${sectionIdentifier}&limitTypeCode=${limitTypeCode}&limitKey=${limitKey}&guarantorPartyIdentifier=${guarantorParty}`,
+      },
+    );
+
+    const { status, body } = await api.post(createFacilityGuaranteeUrl, requestBodyWithMaximumLiabilityToRound);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      facilityIdentifier,
+    });
+    expect(acbsRequestWithRoundedMaximumLiability.isDone()).toBe(true);
+  });
+
+  it(`replaces the effectiveDate with today's date if the specified effectiveDate is in future`, async () => {
+    const requestBodyWithFutureEffectiveDate = [{ ...requestBodyToCreateFacilityGuarantee[0], effectiveDate: '3000-01-02' }];
```

Comment:
Could use `A_FUTURE_DATE_ONLY` here.

Reply:
Changed to TEST_DATES.A_FUTURE_EXPIRY_DATE_ONLY
These constant could have more generic names. Maybe: 
A_FUTURE_EFFECTIVE_DATE_ONLY -> A_FUTURE_START_DATE_ONLY
A_FUTURE_EXPIRY_DATE_ONLY -> A_FUTURE_END_DATE_ONLY

Reply:
Created TODO and task ticket APIM-248.